components/src/core/components/Drawer/Drawer.vue

```diff
@@ -0,0 +1,234 @@
+<template>
+  <div class="drawer-wrapper">
+    <div v-if="modalState" class="overlay"></div>
+    <div
+      class="oxd-drawer"
+      ref="wrapper"
+      :class="[
+        alignments,
+        {'sticky-footer': stickyFooter},
+        {opened: modalState},
+      ]"
+      :style="[styles, drawerPositionX]"
+      v-click-outside="clickOutside"
+    >
+      <div class="drawer-header" ref="header">
+        <slot name="header">
+          <h5 v-text="title"></h5>
+        </slot>
+      </div>
+      <div class="drawer-body" :style="bodyHeight">
+        <slot name="body"></slot>
+      </div>
+      <div class="drawer-footer" ref="footer">
+        <slot name="footer">
+          <oxd-button
+            class="default-btn--cancel"
+            :label="cancelButtonData.label"
+            :iconName="cancelButtonData.iconName"
+            :size="cancelButtonData.size"
+            :style="cancelButtonData.style"
+            :displayType="cancelButtonData.displayType"
+            @click="cancelButtonData.click"
+          />
+          <oxd-button
+            class="default-btn--save"
+            :label="okButtonData.label"
+            :iconName="okButtonData.iconName"
+            :size="okButtonData.size"
+            :style="okButtonData.style"
+            :displayType="okButtonData.displayType"
+            @click="okButtonData.click"
+          />
+        </slot>
+      </div>
+      <div v-if="false" class="trigger">
+        <slot name="trigger"></slot>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import {defineComponent, computed, onMounted, ref} from 'vue';
+import Button from '@orangehrm/oxd/core/components/Button/Button.vue';
+import clickOutsideDirective from '@orangehrm/oxd/directives/click-outside';
+
+export default defineComponent({
+  components: {
+    'oxd-button': Button,
+  },
+
+  props: {
+    autoAlign: {
+      type: Boolean,
+      default: true,
+    },
+    position: {
+      type: String,
+      default: 'right',
+    },
+    width: {
+      type: String,
+      default: null,
+    },
+    height: {
+      type: String,
+      default: null,
+    },
+    fullHeight: {
+      type: Boolean,
+      default: false,
+    },
+    stickyFooter: {
+      type: Boolean,
+      default: false,
+    },
+    fixed: {
+      type: Boolean,
+      default: false,
+    },
+    title: {
+      type: String,
+      default: null,
+    },
+    okButton: {
+      type: Object,
+      default: () => null,
+    },
+    cancelButton: {
+      type: Object,
+      default: () => null,
+    },
+    modalState: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
+  directives: {
+    'click-outside': clickOutsideDirective,
+  },
+
+  setup(props, {emit}) {
+    const open = ref(false);
```

Comment:
all these refs can be taken to a [reactive](https://v3.vuejs.org/guide/reactivity-fundamentals.html#declaring-reactive-state) with types.

Reply:
fixed